examples/window-app/silabs/src/WindowManager.cpp

```diff
@@ -0,0 +1,1032 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <AppConfig.h>
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/identify-server/identify-server.h>
+#include <app/server/Server.h>
+#include <app/util/af.h>
+#include "AppEvent.h"
+#include "AppTask.h"
+
+#include <lib/support/CodeUtils.h>
+#include <platform/CHIPDeviceLayer.h>
+
+#include <WindowManager.h>
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/window-covering-server/window-covering-server.h>
+#include <app/server/OnboardingCodesUtil.h>
+#include <lib/core/CHIPError.h>
+#include <lib/dnssd/Advertiser.h>
+#include <lib/support/CodeUtils.h>
+#include <platform/CHIPDeviceLayer.h>
+#ifdef QR_CODE_ENABLED
+#include <qrcodegen.h>
+#else
+#include "SilabsDeviceDataProvider.h"
+#include <setup_payload/QRCodeSetupPayloadGenerator.h>
+#include <setup_payload/SetupPayload.h>
+#endif // QR_CODE_ENABLED
+
+#include <sl_system_kernel.h>
+
+#ifdef SL_WIFI
+#include "wfx_host_events.h"
+#include <app/clusters/network-commissioning/network-commissioning.h>
+#include <platform/silabs/NetworkCommissioningWiFiDriver.h>
+#endif
+
+#ifdef DISPLAY_ENABLED
+#include <LcdPainter.h>
+SilabsLCD slLCD;
+#endif
+
+#include <platform/silabs/platformAbstraction/SilabsPlatform.h>
+
+#define EXAMPLE_VENDOR_ID 0xcafe
+
+#define LCD_ICON_TIMEOUT 1000
+
+using namespace chip::app::Clusters::WindowCovering;
+using namespace chip;
+using namespace ::chip::DeviceLayer;
+using namespace ::chip::DeviceLayer::Silabs;
+#define APP_STATE_LED 0
+#define APP_ACTION_LED 1
+
+#ifdef SL_WIFI
+chip::app::Clusters::NetworkCommissioning::Instance
+    sWiFiNetworkCommissioningInstance(0 /* Endpoint Id */, &(chip::DeviceLayer::NetworkCommissioning::SlWiFiDriver::GetInstance()));
+#endif
+
+using namespace ::chip::Credentials;
+using namespace ::chip::DeviceLayer;
+using namespace chip::app::Clusters::WindowCovering;
+
+WindowManager WindowManager::sInstance;
+
+AppEvent CreateNewEvent(AppEvent::AppEventTypes type){
+    AppEvent aEvent;
+    aEvent.Type = type;
+    aEvent.Handler = WindowManager::GeneralEventHandler;
+    WindowManager * window = static_cast<WindowManager *>(&WindowManager::sInstance);
+    aEvent.WindowEvent.Context = window;
+    return aEvent;
+}
+
+inline void OnTriggerEffectCompleted(chip::System::Layer * systemLayer, void * appState)
+{
+    AppEvent event = CreateNewEvent(AppEvent::kEventType_WinkOff);
+    AppTask::GetAppTask().PostEvent(&event);
+}
+
+void WindowManager::Timer::Start()
```

Comment:
If I'm not mistaken the timer is actually used twice. and we seems to only use a single timer. That doesn't seems to justify all of this wrapping. Please removed appropriately